avm/res/network/private-endpoint/main.bicep

```diff
@@ -23,6 +23,9 @@ param privateDnsZoneGroupName string?
 @description('Optional. The private DNS zone groups to associate the private endpoint. A DNS zone group can support up to 5 DNS zones.')
 param privateDnsZoneResourceIds array?
 
+@description('Optional. Array of names for each private private DNS zone configuration, in the same order as `privateDnsZoneResourceIds`. Individual values can be set to `null` to use the default name.')
+param privateDnsZoneConfigurationNames array?
```

Comment:
Is that a string array like `stringp[]`? Or does it contain objects?

Also, I may suggest (if @segraef agrees) to change the interface from 2 string arrays (`privateDnsZoneConfigurationNames` & `privateDnsZoneResourceIds`) to an object array of private DNS configurations - fixing the issue with 'you have to provide them in the exact same order' etc. to begin with:

`/private-dns-zone-group/main.bicep`
```bicep
@description('Optional. The name of the private DNS zone group.')
param name string = 'default'

@description('Required. Array of private DNS zone resource IDs. A DNS zone group can support up to 5 DNS zones.')
@minLength(1)
@maxLength(5)
param privateDnsZoneConfigs privateDnsZoneGroupConfigType[]

var privateDnsZoneConfigsVar = [
  for privateDnsZoneConfig in privateDnsZoneConfigs: {
    name: privateDnsZoneConfig.?name ?? last(split(privateDnsZoneConfig.privateDnsZoneResourceId, '/'))
    properties: {
      privateDnsZoneId: privateDnsZoneConfig.privateDnsZoneResourceId
    }
  }
]

resource privateEndpoint 'Microsoft.Network/privateEndpoints@2023-11-01' existing = {
  name: privateEndpointName
}

resource privateDnsZoneGroup 'Microsoft.Network/privateEndpoints/privateDnsZoneGroups@2023-11-01' = {
  name: name
  parent: privateEndpoint
  properties: {
    privateDnsZoneConfigs: privateDnsZoneConfigsVar
  }
}

// (...)

// ================ //
// Definitions      //
// ================ //

@export()
type privateDnsZoneGroupConfigType = {
  @description('Optional. The name of the Private DNS Zone Group Config.')
  name: string?

  @description('Required. The Private DNS Zone Group to associate the private endpoint.')
  privateDnsZoneResourceId: string
}
```

`main.bicep`
```bicep
@description('Optional. The private DNS zone group to configure for the private endpoint.')
param privateDnsZoneGroup privateDnsZoneGroupType?

// (...)

module privateEndpoint_privateDnsZoneGroup 'private-dns-zone-group/main.bicep' = if (!empty(privateDnsZoneGroup)) {
  name: '${uniqueString(deployment().name)}-PrivateEndpoint-PrivateDnsZoneGroup'
  params: {
    privateEndpointName: privateEndpoint.name
    name: privateDnsZoneGroup.?name
    privateDnsZoneConfigs: privateDnsZoneGroup!.privateDnsZoneGroupConfigs
  }
}

// (...)

import { privateDnsZoneGroupConfigType } from 'private-dns-zone-group/main.bicep'
type privateDnsZoneGroupType = {
  @description('Optional. The name of the Private DNS Zone Group.')
  name: string?

  @description('Required. The private DNS zone groups to associate the private endpoint. A DNS zone group can support up to 5 DNS zones.')
  privateDnsZoneGroupConfigs: privateDnsZoneGroupConfigType[]
}
```

Reply:
It's an array of strings, however I used `array` so that you can supply `null` in case you don't want to provide your own name. Ideally I would have done something like `(string | null)[]?` but that doesn't seem to work.

As for using an array of objects, I would love to however, I was a bit hesitant about breaking changes. If @segraef agrees I will happily change.

Reply:
Makes sense. 
And yes, let's see. It would nicely align with how AVM modules are structured in general, but it's ultimately up to the module's owner. Please note that Mr. @segraef may or may not be located in a TZ far far away so I don't think we'll hear anything back before tomorrow.

Reply:
> Makes sense. And yes, let's see. It would nicely align with how AVM modules are structured in general, but it's ultimately up to the module's owner. Please note that Mr. @segraef may or may not be located in a TZ far far away so I don't think we'll hear anything back before tomorrow.

<img width="300" alt="image" src="https://github.com/user-attachments/assets/f0c70e92-b2b0-4775-bc56-62ee632ea87b">

Let me have a look and I will come back to you.

Reply:
Note that if we were to implement the change, we must also update the AVM spec after & each module using the new PE version would need to be updated:
https://github.com/Azure/bicep-registry-modules/blob/3f2ddd8e8ba96d0d1d0c5f7376168fa4e2c5bf7a/avm/res/key-vault/vault/tests/e2e/waf-aligned/main.test.bicep#L125

Still, I'd argue it's the correct thing to do.

Reply:
Hi @segraef - What is your view on this?

Reply:
Hello @segraef , sorry to bother, do you have an estimate on when you might be able to look into this? 

The reason we are asking is because we are trying to finish the new `avm/ptn/ai-platform/baseline` module so we can use it in a project, and this PR is currently blocking the completion of that module.

Thank you

Reply:
Thanks for your contribution @cecheta and @AlexanderSehr, much appreciated for your efforts here.
Had a look at it and the idea to avoid a consumer to follow the exact order makes sense. 
Let's go with the array of objects, can you make sure to include this in the e2e @cecheta ?

What's the requirement of having `privateDnsZoneConfigurationNames` configurable for `avm/ptn/ai-platform/baseline`?

Reply:
Sure, thanks for the response, will update.

As for why this is needed, we have been creating AI Studio projects and hubs via Bicep. When a hub has a private endpoint, and you create a project from that hub, a CNAME DNS record for the new project should be automatically be created in the private DNS zone. However, through trial and error, we discovered that that record only gets created if the `privateDnsZoneConfigs` have a specific name, which is the name of the zone with `.` replaced with `-`.

Reply:
@segraef PR has been updated and pipeline re-run

@AlexanderSehr if this is approved and merged I will raise issues for updating AVM and all modules using this

Reply:
Will check again as soon as I can